src/functional/iterator/transforms/global_tmps.py

```diff
@@ -290,6 +292,7 @@ def update_cartesian_domains(
     domains = dict[str, ir.Expr]()
     for closure in reversed(node.fencil.closures):
         if closure.domain == AUTO_DOMAIN:
+            assert isinstance(closure.output, ir.SymRef)
```

Comment:
We need to check how to fix that, because closure.output can be a FunCall. That looks like a bug in this pass.